sklearn/metrics/_classification.py

```diff
@@ -1776,6 +1978,126 @@ def recall_score(y_true, y_pred, *, labels=None, pos_label=1, average='binary',
     return r
 
 
+def specificity_score(y_true, y_pred, *, labels=None, pos_label=1,
+                      average='binary', sample_weight=None,
+                      zero_division="warn"):
+    """Compute specificity, also known as true negative rate.
+
+    The specificity is the ratio ``tn / (tn + fp)`` where ``tn`` is the number
+    of true negatives and ``fp`` is the number of false positives.
+    The specificity is intuitively the ability of the classifier to find
+    all the negative samples.
+
+    The best value is 1 and the worst value is 0.
```

Comment:
We will need to add some documentation in the user guide and linked it.

Reply:
We should also add the sphinx directive:

```python
.. versionadded:: 1.1
```

Reply:
Done.